agent/agent.go

```diff
@@ -319,7 +327,21 @@ func main() {
 	waitForHealthy()
 	go runHealthChecks()
 
-	if err := runAdapter(ctx); err != nil {
-		log.Fatal(err.Error())
+	requestPollingCtx, requestPollingCancel := context.WithCancel(ctx)
+
+	go func(ctx context.Context) {
+		if err := runAdapter(ctx, requestPollingCtx); err != nil {
+			log.Fatal(err.Error())
+		}
+	}(ctx)
+
+	osShutdownSignalCh := utils.ShutdownSignalChan()
+	<-osShutdownSignalCh
+
+	if *gracefulShutdownTimeout > 0 {
+		requestPollingCancel()
+		log.Println("Begin graceful shutdown wait timer.")
```

Comment:
Uber-nit: can we log how long we're about to wait here, to make it easier to debug later?

Reply:
Done.